assets/opengrep_rules/client/service-worker-origin-validation.yaml

```diff
@@ -0,0 +1,39 @@
+rules:
+  - id: chromium-service-worker-origin-validation
+    metadata:
+      author: Andrea Brancaleoni <abc@pompel.me>
+      references:
+        - https://chromium.googlesource.com/chromium/src/+/main/docs/security/service-worker-security-faq.md
+        - https://github.com/brave/brave-browser/wiki/Security-reviews
+      source: https://github.com/brave/security-action/blob/main/assets/opengrep_rules/client/service-worker-origin-validation.yaml
+      assignees: |
+        thypon
+        diracdeltas
+        bridiver
+      category: security
+    languages: [cpp, c]
+    message: |
+      Service Worker operations via ServiceWorkerContext must validate origins to prevent
+      cross-origin attacks. Service Workers have elevated privileges and must
+      enforce same-origin policy strictly. Validate origins before registration or message passing.
+    severity: WARNING
+    patterns:
+      - patterns:
+          - pattern-either:
+              - pattern: $CTX->RegisterServiceWorker($URL, ...)
+              - pattern: $CTX->StartServiceWorker($URL, ...)
```

Comment:
this is an internal method, the only publicly accessible methods are in ServiceWorkerContext